ibis/backends/base/sql/compiler/select_builder.py

```diff
@@ -500,8 +501,11 @@ def _visit_select_Histogram(self, expr):
             binwidth = op.binwidth
             base = op.base
 
-        bucket = (op.arg - base) / binwidth
-        return bucket.floor().name(expr._name)
+        bucket = ((op.arg - base) / binwidth).floor()
+        if expr.has_name():
+            bucket = bucket.name(expr.get_name())
```

Comment:
It looks like there are a bunch of these `has_name` checks that aren't tested.

Reply:
Breaks tests if I remove the `has_name()` check.